### ISSUE TITLE
Bug: Grafana 4.3.0 not sending pictures to proper s3 url

### ISSUE THREAD

**Issue by mnuic** [grafana/grafana#8444]
Please include this information:
- What Grafana version are you using?
  4.3.0
- What datasource are you using?
  graphite, cloudwatch, etc...
- What OS are you running grafana on?
  docker image
- What did you do?
  download latest grafana image 4.3.0. When sending alert with pictures, pictures are stored in s3 that work in version 4.2.0, but not in 4.3.0

defined link in defaults.ini is:
   https://grafana.s3.amazonaws.com  (works on version 4.2.0)

but insted grafana is sending to:
   https://grafana.**s3-us-east-1**.amazonaws.com (in 4.3.0)
 
I have not specified .s3-us-east-1. in defults.ini configuration.

I have downgraded to docker image 4.2.0 for now.

**Comment by torkelo** [comment]
your s3 url needs to contain s3 region

**Comment by torkelo** [comment]
can you update an test if that works?

**Comment by mnuic** [comment]
Offical documentation: http://docs.grafana.org/installation/configuration/
```
bucket_url¶
Bucket URL for S3. AWS region can be specified within URL or defaults to ‘us-east-1’, e.g. - http://grafana.s3.amazonaws.com/ - https://grafana.s3-ap-southeast-2.amazonaws.com/ - https://grafana.s3-cn-north-1.amazonaws.com.cn
```


When I specify s3 region:
bucket_url = https://grafana.s3-us-east-1.amazonaws.com

```
t=2017-05-23T14:45:35+0000 lvl=info msg="Sending notifications for" logger=alerting.notifier ruleId=7 sent count=1
t=2017-05-23T14:45:35+0000 lvl=info msg=Rendering logger=png-renderer path="XYZpanelId=3"
t=2017-05-23T14:45:37+0000 lvl=info msg=uploaded logger=alerting.notifier url=https://grafana.s3-us-east-1.amazonaws.com/cEJHlGTO3sgGGxeTMN7Y.png
t=2017-05-23T14:45:37+0000 lvl=info msg="Sending notification" logger=alerting.notifier type=slack id=4 isDefault=false
t=2017-05-23T14:45:37+0000 lvl=info msg="Executing slack notification" logger=alerting.notifier.slack ruleId=7 notification=#slack-test
t=2017-05-23T14:45:45+0000 lvl=info msg="New state change" logger=alerting.resultHandler alertId=7 newState=alerting prev state=ok
t=2017-05-23T14:45:45+0000 lvl=eror msg="Failed to save state" logger=alerting.resultHandler error="Could not find alert"
```


The picture is not even save on s3 on grafana 4.3.0.

This works on grafana 4.2.0

**Comment by torkelo** [comment]
ok, thanks for the info will investigate and try to replciate!

**Comment by torkelo** [comment]
very strange that this us-east-1 region is handled so differently, every other region uses `s3-<region>.amazonaws.com` in the domain, but us standard region only works with no region defined in the domain.

**Comment by mnuic** [comment]
Yes, that is right. It confused me too, but [aws documents](http://docs.aws.amazon.com/general/latest/gr/rande.html#s3_region) is clear.
Maybe this would be more clearer to see if defined url in `defaults.ini` should be used to send to s3.
Example if I write `http://xyz.s3.domain.com` it should write this in log, and not `https://xyz.s3-region-1.domain.com`. 

Second thing I noticed that sometimes defined url `http`, changes in log entries to `https`. Not know if this is related to.

**Comment by torkelo** [comment]
new docker image for 4.3.1 released, thanks again for reporting this :)

**Comment by mnuic** [comment]
Thanks Torkelo, it works now, 4.3.1 :)
